client/client-multi/client-c/src/kaa/kaa_user.c

```diff
@@ -77,6 +93,26 @@ typedef enum {
 static kaa_service_t user_sync_services[1] = {KAA_SERVICE_USER};
 
 
+static void dtor_endpoint_info(void *data)
+{
+    KAA_RETURN_IF_NIL(data, );
+    kaa_endpoint_info_t *endpoint_info = (kaa_endpoint_info_t*)data;
+    if(endpoint_info->access_token)
+        KAA_FREE(endpoint_info->access_token);
+    KAA_FREE(endpoint_info);
+}
+
+static bool match_predicate_endpoint_info(void *data, void *context)
+{
+    KAA_RETURN_IF_NIL(data, false);
+
+    kaa_endpoint_info_t *endpoint_item = (kaa_endpoint_info_t*)data;
+    uint16_t request_id = *(uint16_t*)context;
+
+    if (request_id == endpoint_item->request_id)
+        return true;
+    return false;
```

Comment:
To reduce the amount of code, you may use
"...return request_id == endpoint_item->request_id;"